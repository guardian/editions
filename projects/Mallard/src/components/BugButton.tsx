import React from 'react'
import { useContext } from 'react'
import { createMailtoHandler } from 'src/helpers/diagnostics'
import { Button } from './button/button'
import { isInBeta } from 'src/helpers/release-stream'
import { StyleSheet } from 'react-native'
import { AccessContext } from 'src/authentication/AccessContext'
import { useApolloClient } from '@apollo/react-hooks'

const styles = StyleSheet.create({
    button: {
        position: 'absolute',
        bottom: 20,
        right: 20,
        zIndex: 9999,
    },
})

const BugButton = () => {
    const { attempt } = useContext(AccessContext)
    const client = useApolloClient()
    return isInBeta() ? (
        <Button
            accessibilityLabel="Report a bug button"
            accessibilityHint="Opens a dialog asking if you want to include diagnostic information to your report"
            style={styles.button}
            onPress={createMailtoHandler(client, 'Report a bug', '', attempt)}
            alt="Report a bug"
            icon="î‚š"
        />
    ) : null
}

export { BugButton }
