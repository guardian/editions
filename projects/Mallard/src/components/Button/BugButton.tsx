import React from 'react'
import { Button } from './Button'
import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
    button: {
        position: 'absolute',
        bottom: 20,
        right: 20,
        zIndex: 9999,
    },
})

const BugButton = ({ onPress }: { onPress: () => void }) => (
    <Button
        accessibilityLabel="Report a bug button"
        accessibilityHint="Opens a dialog asking if you want to include diagnostic information to your report"
        style={styles.button}
        onPress={onPress}
        alt="Report a bug"
        icon="î‚š"
    />
)

export { BugButton }
