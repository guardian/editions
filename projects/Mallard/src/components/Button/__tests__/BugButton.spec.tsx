import React from 'react';
import type { ReactTestRendererJSON } from 'react-test-renderer';
import TestRenderer from 'react-test-renderer';
import { BugButton } from '../BugButton';

describe('BugButton', () => {
	it('should display a BugButton', () => {
		const component: ReactTestRendererJSON | null = TestRenderer.create(
			<BugButton onPress={() => {}} />,
		).toJSON();
		expect(component).toMatchSnapshot();
	});
});
