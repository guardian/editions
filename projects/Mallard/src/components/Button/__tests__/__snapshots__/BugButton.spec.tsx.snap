// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BugButton should display a BugButton 1`] = `
<View
  accessibilityHint="Opens a dialog asking if you want to include diagnostic information to your report"
  accessibilityLabel="Report a bug button"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "bottom": 20,
      "opacity": 1,
      "position": "absolute",
      "right": 20,
      "zIndex": 9999,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 999,
          "height": 42,
          "justifyContent": "center",
          "paddingHorizontal": 21,
        },
        {
          "backgroundColor": "#ffe500",
        },
        {
          "alignItems": "center",
          "aspectRatio": 1,
          "justifyContent": "center",
          "paddingHorizontal": 0,
          "width": 42,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        [
          {
            "fontFamily": "GuardianIcons-Regular",
            "fontSize": 20,
            "lineHeight": 20,
          },
          [
            {
              "color": "#121212",
            },
            undefined,
          ],
        ]
      }
    >
      î‚š
    </Text>
  </View>
</View>
`;
